Mathlib/MeasureTheory/Measure/Haar/DistribChar.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2024 Andrew Yang, Yaël Dillies, Javier López-Contreras. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang, Yaël Dillies, Javier López-Contreras
+-/
+import Mathlib.MeasureTheory.Measure.Haar.Unique
+
+/-!
+# The distributive character of Haar measures
+
+Given a group `G` acting by additive morphisms on a locally compact additive commutative group `A`,
+and an element `g : G`, one can pull back the Haar measure `μ` of `A`
+along the map `(g • ·) : A → A` to get another Haar measure `μ'` on `A`.
+By unicity of Haar measures, there exists some nonnegative real number `r` such that `μ' = r • μ`.
+We can thus define a map `distribHaarChar : G → ℝ≥0` sending `g` to its associated real number `r`.
+Furthermore, this number doesn't depend on the Haar measure `μ` we started with,
+and `distribHaarChar` is a group homomorphism.
+
+## See also
+
+`MeasureTheory.Measure.modularCharacter` for the analogous definition when the action is
+multiplicative instead of distributive.
+
+[Zulip](https://leanprover.zulipchat.com/#narrow/channel/217875-Is-there-code-for-X.3F/topic/canonical.20norm.20coming.20from.20Haar.20measure/near/480050592)
+-/
+
+open MeasureTheory.Measure
+open scoped NNReal Pointwise ENNReal
+
+namespace MeasureTheory
+variable {G A : Type*} [Group G] [AddCommGroup A] [DistribMulAction G A] [TopologicalSpace A]
+  [IsTopologicalAddGroup A] [LocallyCompactSpace A] [ContinuousConstSMul G A] {g : G}
+
+variable (A) in
+/-- The distributive Haar character of a group `G` acting distributively on a group `A` is the
+unique positive real number `Δ(g)` such that `μ (g • s) = Δ(g) * μ s` for all Haar
+measures `μ : Measure A`, set `s : Set A` and `g : G`. -/
+@[simps -isSimp]
+noncomputable def distribHaarChar : G →* ℝ≥0 :=
+  letI := borel A
+  haveI : BorelSpace A := ⟨rfl⟩
+  {
+    toFun g := addHaarScalarFactor (DomMulAct.mk g • addHaar) (addHaar (G := A))
+    map_one' := by simp
+    map_mul' g g' := by
+      simp_rw [DomMulAct.mk_mul]
+      rw [addHaarScalarFactor_eq_mul _ (DomMulAct.mk g' • addHaar (G := A))]
+      congr 1
+      simp_rw [mul_smul]
+      rw [addHaarScalarFactor_domSMul]
+  }
+
+lemma distribHaarChar_pos : 0 < distribHaarChar A g :=
+  pos_iff_ne_zero.mpr ((Group.isUnit g).map (distribHaarChar A)).ne_zero
+
+variable [MeasurableSpace A] [BorelSpace A] {μ : Measure A} [μ.IsAddHaarMeasure]
+
+variable (μ) in
+lemma addHaarScalarFactor_smul_eq_distribHaarChar (g : G) :
+    addHaarScalarFactor (DomMulAct.mk g • μ) μ = distribHaarChar A g := by
+  obtain ⟨rfl⟩ := ‹BorelSpace A›
+  letI := borel A
```

Comment:
```suggestion
  borelize A
```

Reply:
Oh and I felt smart coming up with those two lines 😅